sites/singleton.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABCMeta
+
+
+class SingletonABCMeta(ABCMeta):
```

Comment:
❓ Interesting! Just wanna make sure: The way I understand it is that if, for example, the `WashingtonCityPaper` class inherits from the singleton abstract `Site` class, there will always be, at maximum, just _one_ instantiation of `WashingtonCityPaper`. Using the singleton pattern here tracks with the fact that we really only need one instantiation to create one `WashingtonCityPaper` object, which we then pass to other parts of the training job.

Am I missing anything?

Reply:
Correct! It actively prevents us from having multiple WCP configurations.

Reply:
Thanks! Would you characterize instantiating a second WCP config elsewhere in the code a bad use, and the preferred way is to pass the config object? If so, is it worth having `SingletonABCMeta.__call__` log a warning or even raise an exception when someone tries to instantiate WCP a second time? If we're setting up a rule that there should only be one WCP, we might as well enforce it.

Reply:
It'd be pretty odd for someone to instantiate another one since we define it right in each site file. The current design is more "just let it work" instead of "break break you're doing a bad thing". If you feel strongly that a warning would be helpful, I'm happy to add it; I don't think it is necessary, but I've also seen a lot more singletons given my Scala programming days!

And the "enforcement" happens by it literally not letting you instantiate a new one anyway.